File: app/src/main/java/com/fitaleks/heroesthesaurus/network/MarvelFetchService.java
package com.fitaleks.heroesthesaurus.network;

import com.fitaleks.heroesthesaurus.data.MarvelCharacter;

import java.util.List;

import retrofit.http.GET;
import retrofit.http.Query;
import rx.Observable;

/**
 * Created by alexander on 21.08.15.
 */
public interface MarvelFetchService {

    @GET("/characters")
    Observable<List<MarvelCharacter>> getCharacters(@Query("orderBy") String orderBy,
                                                    @Query("ts") long currentTime,
                                                    @Query("hash") String hash,
                                                    @Query("offset") long offset);

    @GET("/characters")
    Observable<List<MarvelCharacter>> getCharacters(@Query("orderBy") String orderBy,
                                                    @Query("ts") long currentTime,
                                                    @Query("hash") String hash,
                                                    @Query("offset") long offset,
                                                    @Query("nameStartsWith") String charName);
}

File: app/src/main/java/com/fitaleks/heroesthesaurus/network/MarvelFetchService.kt
package com.fitaleks.heroesthesaurus.network

import com.fitaleks.heroesthesaurus.data.MarvelCharacter
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by alexander on 21.08.15.
 */
interface MarvelFetchService {

    @GET("characters")
    fun getCharacters(@Query("orderBy") orderBy: String,
                      @Query("offset") offset: Long): Call<List<MarvelCharacter>>

    @GET("characters")
    fun getCharacters(@Query("orderBy") orderBy: String,
                      @Query("offset") offset: Long,
                      @Query("nameStartsWith") charName: String): Call<List<MarvelCharacter>>
}
